app/src/main/java/com/mk/tasky/agenda/data/AgendaRepositoryImpl.kt

```diff
@@ -70,25 +71,23 @@ class AgendaRepositoryImpl(
     override fun getAgenda(date: LocalDateTime, forceRemote: Boolean): Flow<Agenda> {
         // TODO: Update with Tasks and Events
         return flow {
-            val localReminders = getRemindersForDate(date.toLocalDate()).toMutableList()
-            emit(Agenda(reminders = localReminders, tasks = emptyList(), events = emptyList()))
+            val localReminders = getRemindersForDate(date.toLocalDate())
+            val localTasks = getTasksForDate(date.toLocalDate())
+            emit(Agenda(localReminders + localTasks))
             if (forceRemote) {
                 getAgendaRemotely(date).onSuccess { response ->
                     supervisorScope {
                         response.reminders.map {
                             launch { dao.insertReminder(it.toDomain().toEntity()) }
                         }.forEach { it.join() }
+                        response.tasks.map {
+                            launch { dao.insertTask(it.toDomain().toEntity()) }
+                        }.forEach { it.join() }
```

Comment:
I think I commented on something like this the last time right? Here, inserting the tasks would start after all reminders have been inserted which can all happen in parallel. Therefore, either use a separate scope or just save references to the jobs and join them all together at the end of the scope.

Reply:
Will do the last one. Grab all the jobs and join them
Thanks!